File: MilkManagementproject/milk_system/src/main/java/com/example/services/MilkProviderService.java
package com.example.services;

import java.util.*;

import com.example.model.*;

public interface MilkProviderService {
	public List<MilkProvider> getAllProviders();
	
	public Optional<MilkProvider> getProviderById(Long providerId);
	
	public MilkProvider addProvider(MilkProvider provider);
	
	 public void deleteProvider(Long providerId);
	 
	 public MilkProvider updateProvider(Long providerId, MilkProvider updatedProvider);
}

File: MilkManagementproject/Frontend_milk/milk_system/src/services/milkproviderService.js
import axios from "axios";

const PROVIDER_API_URL = "http://localhost:8080/api/providers";

export const getAllProviders = () => axios.get(PROVIDER_API_URL);
export const addProvider = (provider) => axios.post(PROVIDER_API_URL, provider);
export const getProviderById = (id) => axios.get(`${PROVIDER_API_URL}/${id}`);
export const updateProvider = (id, provider) => axios.put(`${PROVIDER_API_URL}/${id}`, provider);
export const deleteProvider = (id) => axios.delete(`${PROVIDER_API_URL}/${id}`);